fre/cmor/cmor_mixer.py

```diff
@@ -1,408 +1,596 @@
+                              local_var = None,
+                              netcdf_file = None,
+                              target_var = None,
+                              json_exp_config = None,
+                              json_table_config = None):#, tmp_dir = None            ):
+    ''' rewrite the input netcdf file nc_fl containing target_var in a CMIP-compliant manner.
+    '''
+    print('\n\n-------------------------- START rewrite_netcdf_file_var call -----')
+    print( "(rewrite_netcdf_file_var) input data: " )
+    print(f"(rewrite_netcdf_file_var)     local_var   =  {local_var}" )
+    print(f"(rewrite_netcdf_file_var)     target_var = {target_var}")
 
-    # Check var_dim, vert_dim
-    if var_dim not in [3, 4]:
-        raise ValueError(f"var_dim == {var_dim} != 3 nor 4. stop.")
 
-    if var_dim == 4 and vert_dim not in [ "plev30", "plev19", "plev8",
-                                          "height2m", "level", "lev", "levhalf"] :
-        raise ValueError(f'var_dim={var_dim}, vert_dim = {vert_dim} is not supported')
+    # open the input file
+    print(f"(rewrite_netcdf_file_var) opening {netcdf_file}" )
+    ds = nc.Dataset(netcdf_file,'a')
+
 
+    # ocean grids are not implemented yet.
+    print( '(rewrite_netcdf_file_var) checking input netcdf file for oceangrid condition')
+    check_dataset_for_ocean_grid(ds)
 
-    print(f"(netcdf_var) var_dim = {var_dim}, var_lst[gfdl_var] = {var_j}")
-    print(f"(netcdf_var)  gfdl_var = {gfdl_var}")
-    units = proj_table_vars["variable_entry"] [var_j] ["units"]
-    #units = proj_table_vars["variable_entry"] [gfdl_var] ["units"]
-    print(f"(netcdf_var) var_dim = {var_dim}, units={units}")
 
-    # "figure out the names of this dimension names programmatically !!!"
+    # figure out the dimension names programmatically TODO
+    # Define lat and lon dimensions
+    # Assume input file is lat/lon grid
     lat = ds["lat"][:]
     lon = ds["lon"][:]
     lat_bnds = ds["lat_bnds"][:]
     lon_bnds = ds["lon_bnds"][:]
-    cmor_lat = cmor.axis("latitude", coord_vals = lat, cell_bounds = lat_bnds, units = "degrees_N")
-    cmor_lon = cmor.axis("longitude", coord_vals = lon, cell_bounds = lon_bnds, units = "degrees_E")
 
-    # Define time and time_bnds dimensions
+    ## Define time
     #time = ds["time"][:]
+
+    # read in time_coords + units
     time_coords = ds["time"][:]
     time_coord_units = ds["time"].units
+    print(f"(rewrite_netcdf_file_var) time_coord_units = {time_coord_units}")
+
+    # read in time_bnds , if present
     time_bnds = []
-    print(f"(netcdf_var) time_coord_units = {time_coord_units}")
-    print(f"(netcdf_var) time_bnds  = {time_bnds}")
     try:
-        print( f"(netcdf_var) Executing cmor.axis('time', \n"
-               f"(netcdf_var) coord_vals = \n{time_coords}, \n"
-               f"(netcdf_var) cell_bounds = {time_bnds}, units = {time_coord_units})   " )
-
         time_bnds = ds["time_bnds"][:]
```

Comment:
This is a non-change, but just noticed it. It's not a problem but I'll point it out since you like netcdf/cf standards now :)

Recall how you are grabbing the dimension names with the CF standards `axis`?

time bounds is actually the same, but by convention it's usually called "time_bnds". The CF rule is that the "time" variable should have a "bounds" attribute pointing to the name of the time bounds variable.

Reply:
ahhhhh, I see. That's a good idea. 

If you don't mind however, this kind of checking for bounds attributes is going to come up when i'm checking other coordinate/dimension things in PR #218. There, I'm working on a few different conditional-checks to read data in a bit more dynamically.